src/main/java/org/wildfly/security/sasl/util/SaslMechanismInformation.java

```diff
@@ -679,6 +680,43 @@ public static boolean needsServerCredentials(final String mechName) {
         }
     }
 
+    /**
+     * Determine whether the given mechanism name is known to not use any sort of {@link Principal} for authentication.
+     *
+     * @param mechName the mechanism name (must not be {@code null})
+     * @return {@code true} if the mechanism does not use a principal, {@code false} if it does or it is not known
+     */
+    public static boolean doesNotUsePrincipal(final String mechName) {
+        switch (mechName) {
+            case Names.ANONYMOUS:
```

Comment:
Yeah the name isn't really a principal though, it's just a scratchpad.  I was thinking even of changing it to a general text input callback with a default set.  But, I think this is OK for the moment.